lighthouse-core/gather/gatherers/image-elements.js

```diff
@@ -308,30 +312,39 @@ class ImageElements extends Gatherer {
 
     /** @type {Array<LH.Artifacts.ImageElement>} */
     const imageUsage = [];
-    const top50Images = Object.values(indexedNetworkRecords)
-      .sort((a, b) => b.resourceSize - a.resourceSize)
-      .slice(0, 50);
+
     await Promise.all([
       driver.sendCommand('DOM.enable'),
       driver.sendCommand('CSS.enable'),
       driver.sendCommand('DOM.getDocument', {depth: -1, pierce: true}),
     ]);
 
+    // Sort (in-place) as largest images descending
+    elements.sort((a, b) => {
+      const aRecord = indexedNetworkRecords[a.src] || {};
+      const bRecord = indexedNetworkRecords[b.src] || {};
+      return bRecord.resourceSize - aRecord.resourceSize;
+    });
+
+    // Don't allow more than 3s of this expensive protocol work
+    let hasBudget = true;
+    setTimeout(_ => (hasBudget = false), 3000);
```

Comment:
Can this be configurable?

Reply:
via LH config? it could.. but it seems early for that. even DEFAULT_PROTOCOL_TIMEOUT isn't exposed to LH.Config

Reply:
I meant an audit option

Reply:
This would need to be a gatherer option though, which I finally removed after several years of non-use 😆 

They can come back soon in FR!

Reply:
Ah, ok, nevermind :)